bgpd/bgp_snmp_bgp4v2.c

```diff
@@ -439,45 +439,79 @@ bgp4v2PathAttrLookup(struct variable *v, oid name[], size_t *length,
 	unsigned int len;
 	struct ipaddr paddr = {};
 	size_t namelen = v ? v->namelen : BGP4V2_NLRI_ENTRY_OFFSET;
-	sa_family_t family = name[namelen - 1] == 1 ? AF_INET : AF_INET6;
-	afi_t afi = AFI_IP;
-	size_t afi_len = IN_ADDR_SIZE;
+	sa_family_t family;
+	sa_family_t min_family = 0; /* family of the selected min path */
+	afi_t afi;
+	safi_t safi;
+	size_t afi_len;
+	long prefix_type = 0;
+	long peer_addr_type = 0;
+
+	/* Bgp4V2AddressFamilyIdentifierTC limited to IPv6 */
+	if (name[namelen - 1] > IANA_AFI_IPV6)
+		return NULL;
+	afi = afi_iana2int(name[namelen - 1]);
+	afi_len = afi == AFI_IP ? IN_ADDR_SIZE : IN6_ADDR_SIZE;
 
-	if (family == AF_INET6) {
-		afi = AFI_IP6;
-		afi_len = IN6_ADDR_SIZE;
-	}
+#define BGP_NLRI_ENTRY_OFFSET namelen
 
-#define BGP_NLRI_ENTRY_OFFSET (afi_len + 1 + afi_len)
 
 	sockunion_init(&su);
 
 	if (exact) {
 		if (*length - namelen != BGP_NLRI_ENTRY_OFFSET)
 			return NULL;
 
-		/* Set OID offset for prefix */
+		/* Set OID offset for prefix type */
 		offset = name + namelen;
-		if (family == AF_INET)
-			oid2in_addr(offset, afi_len, &addr->u.prefix4);
-		else
+
+		/* Bgp4V2SubsequentAddressFamilyIdentifierTC  */
+		/* limited to Labeled unicast */
+		if (*offset > IANA_SAFI_LABELED_UNICAST)
+			return NULL;
+		safi = safi_iana2int(*offset);
+		offset++;
+
+		/* get bgp4V2NlriPrefixType */
+		prefix_type = *offset;
+		offset++;
+
+		/* get bgp4V2NlriPrefix */
+		if (prefix_type == INET_ADDR_TYPE_IPV4) {
```

Comment:
We already have `iana_afi_t`, can we use this enum for `prefix_type` and drop `INET_ADDR_TYPE_IPV4`, `INET_ADDR_TYPE_IPV6` from a header file?

Reply:
done